### ISSUE TITLE
Bug: Critical damage formula

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#580]
In GitLab by @aestrada4355 on Oct 30, 2021, 04:52

When using the Extra Critical Hit Damage field from core 5e and rolling a critical hit, the Extra die are incorrectly added and multiplied.

The expected damage formula output is 2d12 + 2 + 1d4, however, the actual output is as seen below.

[fvtt-midi-qol-settings.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/29ea9506d6ef879a7004a62507ebcc856e64805b46b30225bedb12b9d4809af4/fvtt-midi-qol-settings.json)

![image](https://my-gitlab-bucket.s3.amazonaws.com/640711674/ccba4bf11aad5705eef7c1f6c5dd7bb3cc1d37122f5367add20ce9410349d6eb/image.png)

**Comment by tposney** [comment]
Your settings have critical damage set to "dnd5e default", which means midi-qol does not do anything to alter the damage rolled by the core system so something else is going on.

Can you export the actor to json for me.

I would guess that you might have a damage bonus defined for the character, but I'll be able to see what's happening if I get the actor.

**Comment by tposney** [comment]
In GitLab by @aestrada4355 on Nov 1, 2021, 23:22

[fvtt-Actor-merric-_halfling-barbarian_.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/bf4dd725b3d36f065a9dab176df63681efba896c703b570467f17cd825e28544/fvtt-Actor-merric-_halfling-barbarian_.json)

Here is the actor used in the test.

Of note, this behavior only occurs when manually rolling damage. If the Midi-qol settings are set to auto-roll damage with auto-fast forward damage enabled the correct damage formula is output.

![image](https://my-gitlab-bucket.s3.amazonaws.com/640711674/3b6aa45f12eb300cda7b5bb363f0ed8b2466a1bb6f161da876bc62b81a361ba1/image.png)
- The 1st entry is manual roll, no fast forward on a critical hit and pressing the Critical Hit button in the damage roll dialogue box. (Observe the issue reported above with the incorrect Extra Critical Hit Damage die being added as 2d4 + 1d4).
- The 2nd entry is manual roll, no auto-fastforward on a critical hit and pressing the Normal button in he damage roll dialogue box. (Observe the weapons damage die is not doubled to 2d12).
- The 3rd entry is automatic damage roll with auto-fastforward on a critical hit. This is the correct and expected output.

**Comment by tposney** [comment]
Thanks for that. 

It's actually a dnd5e bug or, rather, a case that was not considered.
If the critical flag for the damage roll is set to true and the roll is still going via the config dialog, the formula that is used adds the bonus critical dice into the formula, which is incorrect.
```
actor.items.getName("Greataxe").rollDamage({critical: true, configureDialog: true})
```
even with midi not installed reproduces the error. I'll raise it on the dnd5e site.

**Comment by tposney** [comment]
I've logged an issue on the dnd5e gitlab and it's slated for release in dnd5e 1.5.4

**Comment by tposney** [comment]
IF the fix does not appear in 1.5.4 of dnd5e please reopen the issue
